library/core/src/pin.rs

```diff
@@ -195,189 +448,269 @@
 //! But if your combinator contains any other data that does not need to be pinned,
 //! you can make those fields not structural and hence freely access them with a
-//! mutable reference even when you just have <code>[Pin]<[&mut] Self></code> (such as in your own
-//! [`poll`] implementation).
-//!
-//! [Deref]: crate::ops::Deref "ops::Deref"
-//! [`Deref`]: crate::ops::Deref "ops::Deref"
-//! [Target]: crate::ops::Deref::Target "ops::Deref::Target"
-//! [`DerefMut`]: crate::ops::DerefMut "ops::DerefMut"
-//! [`mem::swap`]: crate::mem::swap "mem::swap"
-//! [`mem::forget`]: crate::mem::forget "mem::forget"
-//! [Vec]: ../../std/vec/struct.Vec.html "Vec"
-//! [`Vec::set_len`]: ../../std/vec/struct.Vec.html#method.set_len "Vec::set_len"
-//! [Box]: ../../std/boxed/struct.Box.html "Box"
-//! [Vec::pop]: ../../std/vec/struct.Vec.html#method.pop "Vec::pop"
-//! [Vec::push]: ../../std/vec/struct.Vec.html#method.push "Vec::push"
-//! [Rc]: ../../std/rc/struct.Rc.html "rc::Rc"
-//! [RefCell]: crate::cell::RefCell "cell::RefCell"
-//! [`drop`]: Drop::drop "Drop::drop"
-//! [VecDeque]: ../../std/collections/struct.VecDeque.html "collections::VecDeque"
-//! [`ptr::write`]: crate::ptr::write "ptr::write"
-//! [`Future`]: crate::future::Future "future::Future"
-//! [drop-impl]: #drop-implementation
-//! [drop-guarantee]: #drop-guarantee
-//! [`poll`]: crate::future::Future::poll "future::Future::poll"
-//! [&]: ../../std/primitive.reference.html "shared reference"
-//! [&mut]: ../../std/primitive.reference.html "mutable reference"
-//! [`unsafe`]: ../../std/keyword.unsafe.html "keyword unsafe"
+//! mutable reference even when you just have <code>[Pin]<[`&mut Self`]></code>
+//! (such as in your own [`poll`] implementation).
+//!
+//! [Target]: Deref::Target "ops::Deref::Target"
+//! [`drop`]: Drop::drop "ops::Drop::drop"
+//! [`poll`]: Future::poll "future::Future::poll"
+//!
+//! <!-- These require `alloc`, but we're in `core`. -->
+//! [`std::alloc`]: https://doc.rust-lang.org/std/alloc/index.html
+//! [`Box<T>`]: https://doc.rust-lang.org/std/boxed/struct.Box.html
+//! [`Rc<T>`]: https://doc.rust-lang.org/std/rc/struct.Rc.html
+//! [`Vec<T>`]: https://doc.rust-lang.org/std/vec/struct.Vec.html
+//! [Vec::pop]: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.pop
+//! [Vec::push]: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.push
+//! [`Vec::set_len`]: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.set_len
+//! [`VecDeque<T>`]: https://doc.rust-lang.org/std/collections/struct.VecDeque.html
+//!
+//! [`&mut T`]: &mut
+//! [`&mut self`]: &mut
+//! [`&mut Self`]: &mut
+//! [`&mut Field`]: &mut
+//!
+//! [`Unmovable`]: self-ref
+//! [`Node`]: linked-list
+//! [`List`]: linked-list
+//! [`&mut Node`]: linked-list
+//! [`List::append`]: linked-list
+//!
+//! [&]: https://doc.rust-lang.org//std/primitive.reference.html "shared reference"
+//! [&mut]: https://doc.rust-lang.org//std/primitive.reference.html "mutable reference"
```

Comment:
Please use `reference` instead, which accounts for the nightly channel properly. (`&` and `&mut` would also work, but shortcut links don't allow a custom title AFAIK.)
```suggestion
//! [&]: reference "shared reference"
//! [&mut]: reference "mutable reference"
```